includes/class-dependencychecker.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * DependencyChecker class file.
+ * specific to Newspack functionality.
+ *
+ * @package Newspack\TecnaviaIntegration
+ */
+
+namespace Newspack\TecnaviaIntegration;
+
+// Check if needed functions exists - if not, require them.
+if ( ! function_exists( 'get_plugins' ) || ! function_exists( 'is_plugin_active' ) ) {
+	require_once ABSPATH . 'wp-admin/includes/plugin.php';
+}
+
+/**
+ * Provide functionality to check the plugin's dependencies on other plugin.
+ *
+ * @since 1.0
+ */
+class DependencyChecker {
+
+	const WOOCOMMERCE_PLUGIN              = 'woocommerce/woocommerce.php';
+	const WOOCOMMERCE_MEMBERSHIPS_PLUGIN  = 'woocommerce-memberships/woocommerce-memberships.php';
+	const WOOCOMMERCE_SUBSCRIPTION_PLUGIN = 'woocommerce-subscriptions/woocommerce-subscriptions.php';
+
+	/**
+	 * Check if plugin is installed by getting all plugins from the plugins dir.
+	 *
+	 * @param  string $plugin_slug Slug of the plugin to check for.
+	 * @return bool
+	 */
+	protected static function check_plugin_installed( $plugin_slug ): bool {
```

Comment:
this is a more resources consuming way of doing this. Also, I think the slug might change if the plugin is running on a different directory

A much simpler approach is just to test if one of the main plugin classes exists with `class_exists( 'Newspack\Newspack' )`